homeassistant/components/roborock/image.py

```diff
@@ -146,53 +157,31 @@ def _create_image(self, map_bytes: bytes) -> bytes:
         return img_byte_arr.getvalue()
 
 
-async def create_coordinator_maps(
-    coord: RoborockDataUpdateCoordinator, drawables: list[Drawable]
-) -> list[RoborockMap]:
+async def refresh_coordinators(
```

Comment:
Note: We can revisit how this work in future PRs to add metadata caching and remove the side effects from the image platform, and ensure it runs for all platforms

My recommendation is we need to:
(1) stop platforms from accessing `coordinator.maps` and move them behind some kind of async call for getting map and/or room information
(2) change `get_maps` and `get_rooms` APIs etc to run once like here an expensive call that is loaded once and cached, then refreshed only based on the active map.